examples/opentelemetry/main.go

```
@@ -0,0 +1,292 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	otelprom "go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	greptime "github.com/GreptimeTeam/greptimedb-ingester-go"
+	"github.com/GreptimeTeam/greptimedb-ingester-go/table"
+	"github.com/GreptimeTeam/greptimedb-ingester-go/table/types"
+)
+
+var (
+	metricsBindAddr = ":2233"
+
+	// It connects the OpenTelemetry Collector through local gRPC connection.
+	// You may replace `localhost:4317` with your endpoint.
+	tracingEndpoint = "localhost:4317"
+)
+
+var (
+	// The GreptimeDB address.
+	host = "127.0.0.1"
+
+	// The database name.
+	database = "public"
+)
+
+type client struct {
+	client *greptime.Client
+}
+
+func newClient(tracerProvider trace.TracerProvider, meterProvider *metric.MeterProvider) *client {
+	cfg := greptime.NewConfig(host).WithDatabase(database).
+		WithTraceProvider(tracerProvider).WithTracesEnabled(true).
+		WithMeterProvider(meterProvider).WithMetricsEnabled(true)
+
+	gtClient, err := greptime.NewClient(cfg)
+	if err != nil {
+		log.Fatal(err)
```

You can propagate the error here. For example:

```
func newClient(tracerProvider trace.TracerProvider, meterProvider *metric.MeterProvider) (*client, error) { ... }
```